src/toDo.js

```diff
@@ -0,0 +1,102 @@
+
+    // load the list from the local storage
+    if (localStorage.getItem('list')) {
+      this.list = JSON.parse(localStorage.getItem('list'));
+      this.renderList();
+    }
+
+    // add listener to the buttons
+    this.listInput.addEventListener('keyup', (event) => {
+      if (event.key === 'Enter') {
+        const taskName = this.listInput.value;
+        this.addTask(taskName);
+        localStorage.setItem('list', JSON.stringify(this.list));
+        this.listInput.value = '';
+      }
+    });
+
+    this.toDoList.addEventListener('click', (event) => {
+      if (event.target.tagName === 'IMG' && event.target.classList.contains('list-img')) {
+        const listItem = event.target.closest('li');
+        const editIcon = listItem.querySelector('.list-img');
+        const deleteIcon = listItem.querySelector('.delete-icon');
+        const textElement = listItem.querySelector('.list-paragraph');
+
+        // enable editing option
+        textElement.contentEditable = true;
+        textElement.focus();
+
+        // replace edit icon with delete icon
+        editIcon.style.display = 'none';
+        deleteIcon.style.display = 'inline-block';
+
+        event.target.src = deleteIcons;
+
+        // add event listener to delete icon
+        deleteIcon.addEventListener('click', () => {
+          const index = Array.from(this.toDoList.children).indexOf(listItem);
+          this.removeListItem(index);
+          localStorage.setItem('list', JSON.stringify(this.list));
+        });
+      }
+    });
+
+    this.toDoList.addEventListener('keyup', (event) => {
+      if (event.target.classList.contains('list-paragraph') && event.key === 'Enter') {
+        const listItem = event.target.closest('li');
+        const index = Array.from(this.toDoList.children).indexOf(listItem);
+        const taskName = event.target.innerText.trim();
+        this.updateTask(index, taskName);
+        localStorage.setItem('list', JSON.stringify(this.list));
+      }
+    });
+  }
+
+  removeListItem(index) {
+    this.list.splice(index, 1);
+    this.renderList();
+  }
+
+  updateTask(index, taskName) {
+    if (taskName) {
+      this.list[index].taskName = taskName;
+      this.renderList();
+    }
+  }
+
+  renderList() {
+    this.toDoList.innerHTML = '';
+    this.list.forEach((list, index) => {
+      const listItem = document.createElement('li');
+      listItem.classList.add('list-items');
+      listItem.innerHTML = `
+      <input class='checkbox' type="checkbox">
+      <p class='list-paragraph' contenteditable="true">${list.taskName}</p>
+      <img src=${threedot} alt="#" data-index="${index}" class='list-img'>
+      <img src=${deleteIcons} alt="Delete" class="delete-icon">
+      `;
+      this.toDoList.appendChild(listItem);
+    });
+  }
+
+  addTask(taskName) {
+    taskName = taskName.trim();
+
+    if (!taskName) {
+      return;
+    }
+
+    const newTask = { taskName };
+    this.list.push(newTask);
+    this.renderList();
```

Comment:
According to the milestone's requirements, there should exist a field for the completed state of the todo which should be mapped to a boolean value and also, an index field which would map to the array length at the instance but you currently have only the title in your todo object. Kindly implement the todo object as stated above in order to satisfy the [requirement](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m3_add_remove.md#:~:text=By%20default%20new,set%20to%201)..) for this milestone.

YOUR TODO OBJECT
<img width="259" alt="wrong" src="https://github.com/jubaer919/MV-to-do-list/assets/84930355/b867d210-f98f-4773-950f-4013167896d0">


NOTE: after deleting a todo, the index should automatically be updated to reflet the new state of the array length as you can see in the provided link above